code/game/machinery/machinery.dm

```diff
@@ -361,4 +361,31 @@ Class Procs:
 			I.crit_fail = 1
 		I.loc = loc
 	qdel(src)
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/attack_generic(var/mob/user, var/damage, var/attack_verb, var/wallbreaker)
+	if (istype(user, /mob/living/simple_animal/construct/armoured))

```

Comment:
Thanks for the hint about ex_act()! I was mislead by the /simple_animal/UnarmedAttack proc, which calls the attacked object's attack_generic method.

Amended in 6b87bd3bdd457489aff0573aa6014d3bc9005c14.